app/Main.hs

```diff
@@ -42,10 +42,17 @@ getPlayer playerNumber gameState = case playerNumber of
                                         Player1 -> player1 gameState
                                         Player2 -> player2 gameState 
 
+flagPrompt :: IO Int
+flagPrompt = do
+    input <- ask "Please pick a flag"
+    let result = reads input :: [(Int, String)]
+    case result of 
+        (first:rest) -> return $ fst first
+        _ -> flagPrompt
 
 chooseFlag :: PlayerNumber -> GameState -> IO Flag
-chooseFlag = undefined
-
+chooseFlag playerNumber gameState = toEnum <$> flagPrompt
```

Comment:
This is a partial function because the user could choose something that it outside the range of enum:

```
Please pick a flag
2
Test.hs: toEnum{Flag}: tag (2) is outside of enumeration's range (0,1)
```